### DISCUSSION TITLE
Schedule Supabase functions—pass secret key to function call?

### DISCUSSION THREAD

**Discussion by henningko** [supabase/supabase#9242] [Questions]
Hi,

While scheduling for edge functions has not yet been released (is there an update?), I'm thinking about using the database as a cron server, as outlined here:
https://supabase.com/blog/postgres-as-a-cron-server

However, I'd need to invoke my edge function with the secret key, and I do not want to stick it verbatim into the the `http_post(...)` function. Is there a way to pass in the secret key, akin to how one would use environment variables elsewhere?

Thanks,

Henning

**Comment by GaryAustin1** [comment]
I've not looked into this yet, but maybe... https://supabase.com/blog/supabase-vault

Otherwise having a table in a private schema with parameters you select from your function is about all you can do.  Advantage is then you don't have to touch the function code every time you move/use it on another instance, just update the table and all functions that use it get it.

**Reply by henningko** [reply]
Thank you. It's not ideal, but since it seems that there's a more comprehensive trigger and scheduling solution coming eventually (🤞), it might be good enough to fill the gap.
Here's what I'll likely to:
1) Create a private table that saves API keys (hopefully, in absence of Vault, something that can be done with `pg_sodium`), and a table to store `anon` key.
2) Retrieve key in SQL function, call edge function with this and the `anon` key.
3) In edge function, do an API verification before executing, via another SQL function/RPC.
